src/wallet/wallet.h

```diff
@@ -1065,6 +1077,9 @@ class CWallet final : public WalletStorage, public interfaces::Chain::Notificati
     //! Find the private key for the given key id from the wallet's descriptors, if available
     //! Returns nullopt when no descriptor has the key or if the wallet is locked.
     std::optional<CKey> GetKey(const CKeyID& keyid) const;
+
+    //! Export the descriptors from this wallet so that they can be imported elsewhere
+    util::Result<std::vector<WalletDescInfo>> ExportDescriptors(bool export_private) const EXCLUSIVE_LOCKS_REQUIRED(cs_wallet);
```

Comment:
In 67216700ce8ec5e84bca5f83860091f6ac4f67f4 _wallet: Move listdescriptors retrieving from RPC to CWallet_:

Could try the new `util::Expected<<std::vector<WalletDescInfo>>,std::string>` from #34006 here.

Reply:
Done